app/Policies/ProductPolicy.php

```diff
@@ -11,15 +11,15 @@ class ProductPolicy
     /**
      * Determine whether the user can view any models.
      */
-    public function viewAny(User $user): bool
+    public function viewAny(?User $user): bool
```

Comment:
I didn't realise that you can even do this for a Policy! Nice! My idea was to 'exempt' the index / show actions from the authorizeResource. But this method is actually really cool! Note that you'd need to check for the $user being null inside the function in this instance (can't assume it's non-null).